src/Npgsql/Internal/Resolvers/RangeTypeInfoResolver.cs

```diff
@@ -1,13 +1,58 @@
 using System;
-using Npgsql.Internal;
+using System.Numerics;
+using Npgsql.Internal.Converters;
 using Npgsql.PostgresTypes;
+using NpgsqlTypes;
 
 namespace Npgsql.Internal.Resolvers;
 
 sealed class RangeTypeInfoResolver : IPgTypeInfoResolver
 {
+    TypeInfoMappingCollection Mappings { get; }
+
+    public RangeTypeInfoResolver()
+    {
+        Mappings = new TypeInfoMappingCollection();
+        AddInfos(Mappings);
+        // TODO: Opt-in only
+        AddArrayInfos(Mappings);
+    }
+
     public PgTypeInfo? GetTypeInfo(Type? type, DataTypeName? dataTypeName, PgSerializerOptions options)
+        => Mappings.Find(type, dataTypeName, options);
+
+    static void AddInfos(TypeInfoMappingCollection mappings)
+    {
+        mappings.AddStructType<NpgsqlRange<int>>(DataTypeNames.Int4Range,
```

Comment:
What about the nullable elements?

Reply:
IIRC we don't support that currently... NpgsqlRange manages absent bounds differently, via the flags rather than by having a nullable bound (i.e. {Lower,Upper}BoundInfinite, when both are absent then Empty).

I know there's code in the converter that deals with nulls (mainly because in the old model it was handled within ReadWithLength)... There have also been discussions over the years to redo how we support ranges (including with nullable boundS), but the current NpgsqlRange seemed "good enough" that it wasn't worth the effort and the break.

I say let's keep things as they are for now - we're not breaking/removing any functionality...